### PR TITLE
Add support for richer console colors and FontWeight and Bold TextDecorator

### PR DESCRIPTION
**Pull request by tomlm**
# You can use any colors for brushes
supports full RGB(255,255,255)
```xaml
<TextBlock Foreground="Magenta" Background="LimeGreen">...</TextBlock>
```
# TextDecorations.Underline and FontWeight support

```xaml
<TextBlock TextDecoration="Underline" FontWeight="Bold">...</TextBlock>
```
Colors

Changes:
* Added Crayon library for better control over console commands
* Renamed FourBitColorBrush to ConsoleColorBrush that uses Color instead of ConsoleColor.
* Changed all ConsoleColor references to be Color references
* Cleaned up Pixel a bit
* Changed pixelbuffer to track additional properties for determining when to flush
* Added Colors page to gallery
* Added TextBlock samples to textblock gallery
* Plumbed FontWeight/FontStyle through rendering pipeline
* Cleaned up PixelForeground and PixelBackground
* Added unit tests.
* Fixed bug on click and drag with mouse throwing exception
* Fixed bug with wheel mouse scrolling. #97 

## Summary by CodeRabbit

- **New Features**
	- Introduced `ConsoleBrush` for enhanced color management across various components.
	- Added new user controls for color and gradient selection in the gallery.
	- Enhanced text rendering capabilities with support for font styles and decorations.

- **Bug Fixes**
	- Updated text alignment verification in tests for improved accuracy.

- **Documentation**
	- Improved project file handling for XAML resources.
  
- **Chores**
	- Updated testing framework dependencies for better support.

- **Style**
	- Simplified color definitions by removing custom brush dependencies in UI components.
	- Updated various controls to use `ConsoleBrush` instead of `FourBitColorBrush`.

### CODE REVIEW

**Comment by jinek** [comment]
It looks stunning. I have changed the code inspections minimum severity from `notice` to `warning` (those based on editor config). Please, pull the changes from main branch.

Still, in this branch there is a warning:

```
Error: /home/runner/work/Consolonia/Consolonia/src/Consolonia.Core/Drawing/PixelBufferImplementation/Pixel.cs(75,25): error CA1508: 'simpleSymbol is ISymbol symbol' is always 'true'. Remove or refactor the condition(s) to avoid dead code. [/home/runner/work/Consolonia/Consolonia/src/Consolonia.Core/Consolonia.Core.csproj]
```

In RELEASE configuration all the warnings turn to errors.

**Comment by tomlm** [comment]
I got LinearGradientBrush support working!
image

File: src/Consolonia.Themes.TurboVision/Themes/TurboVisionDark/TurboVisionDarkColors.axaml [hunk 1]
```diff
@@ -2,28 +2,28 @@
...
                                Color="Gray" />
-    <drawing:FourBitColorBrush x:Key="ThemeShadeBrush"
+    <drawing:ConsoleBrush x:Key="ThemeShadeBrush"
                                Mode="Shaded" />
-    <drawing:FourBitColorBrush x:Key="ThemePseudoShadeBrush"
+    <drawing:ConsoleBrush x:Key="ThemePseudoShadeBrush"
                                Color="Black" />
-    <drawing:FourBitColorBrush x:Key="ThemeErrorBrush"
+    <drawing:ConsoleBrush x:Key="ThemeErrorBrush"
                                Color="DarkRed" />
```
[the remark this answers is not in the document]
Comment (tomlm) [reply src/Consolonia.Themes.TurboVision/Themes/TurboVisionDark/TurboVisionDarkColors.axaml:28]: I don' see that I changed the colors. This feels like a false alert so I'm resolving.

File: src/Consolonia.Gallery/Gallery/GalleryViews/GalleryTextBlock.axaml [hunk 2]
```diff
@@ -11,35 +11,37 @@
...
+                       TextAlignment="Center" />
+            <TextBlock Text="Right aligned text"
+                       TextAlignment="Right" />
+        </StackPanel>
+
         <TextBlock TextWrapping="Wrap"
                    Text="Multiline TextBlock with TextWrapping.&#xD;&#xD;Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus magna. Cras in mi at felis aliquet congue. Ut a est eget ligula molestie gravida. Curabitur massa. Donec eleifend, libero at sagittis mollis, tellus est malesuada tellus, at luctus turpis elit sit amet quam. Vivamus pretium ornare est." />
         <TextBlock TextWrapping="Wrap"
                    Text="Special characters: Élève naïve: “𝔉𝔞𝔫𝔠𝔶” ligatures like ﬀ, ﬁ, and ﬂ; numbers ½, ¼; symbols like @, #, and €; emoji 😊; and math symbols Ω, ∑, √…"/>
+        <TextBlock Text="TextDecoration=Underline" TextDecorations="Underline" HorizontalAlignment="Left"/>
```
[the remark this answers is not in the document]
Comment (tomlm) [reply src/Consolonia.Gallery/Gallery/GalleryViews/GalleryTextBlock.axaml:33]: strikethrough doesn't work and proabably never will..

**Comment by jinek** [comment]
@tomlm  This is new one:

>  Error: /home/runner/work/Consolonia/Consolonia/src/Consolonia.Core/Drawing/ConsoleBrush.cs(74,21): error CS0162: Unreachable code detected 

By the way, do you see log of the builds?

**Comment by jinek** [comment]
> Wow, this coderabbit tool is **unbelievable!** 

In a good or bad way? I was thinking to set it to be less verbose/sensitive

**Comment by tomlm** [comment]
In a very good way. I recently retired and one of the things I miss is the valuable feedback I would get from colleagues. The quality of the code review is sooooo much better than crappy linting tools. I'm super impressed!

**Comment by tomlm** [comment]
I *think* I've resolved all of the coderabbit issues, I'm ready to go if the build works and you are satisfied.

**Comment by tomlm** [comment]
huh. Code inspection build is failing and I don't understand why
It's failing to checkout the repo?
image

**Comment by jinek** [comment]
> huh. Code inspection build is failing and I don't understand why
> It's failing to checkout the repo?
> image
> 
> 

May be that because this action does not work with forks. Can you rebase to this repo branch? Or as an option i can run code inspections/cleanup in a separate PR afterwards.

**Comment by tomlm** [comment]
I'll try. I always get a bit nervous doing that.

**Comment by tomlm** [comment]
BTW: I've got another PR I'm working on:
image

**Comment by jinek** [comment]
> BTW: I've got another PR I'm working on:
> image
> 

There was an issue regarding this btw, will find later

**Comment by jinek** [comment]
> I'll try. I always get a bit nervous doing that. 

Im thinking eventually we need to setup the pipeline to properly validate forks as well. But also i was thinking may be better to migrate to qodana instead of jetbrains cli. Thus, if we can, let's do pull requests right in the repo for now.

**Pull request closed without merging**